appIcons.js

```diff
@@ -1740,62 +1740,48 @@ var MyShowAppsIconMenu = Utils.defineClass({
 
     _dtpRedisplay: function() {
         this.removeAll();
+        
+        // Only add menu entries for commands that exist in path
+        function _appendItem(obj, info) {
+            if (Utils.checkIfCommandExists(info.cmd[0])) {
+                let item = obj._appendMenuItem(_(info.title));
+
+                item.connect('activate', function() {
+                    Util.spawn(info.cmd);
+                });
+                return item;
+            }
 
-        if (this.sourceActor != Main.layoutManager.dummyCursor) { 
-            let powerSettingsMenuItem = this._appendMenuItem(_('Power options'));
-            powerSettingsMenuItem.connect('activate', function () {
-                Util.spawn(['gnome-control-center', 'power']);
-            });
-
-            let logsMenuItem = this._appendMenuItem(_('Event logs'));
-            logsMenuItem.connect('activate', function () {
-                Util.spawn(['gnome-logs']);
-            });
-
-            let systemSettingsMenuItem = this._appendMenuItem(_('System'));
-            systemSettingsMenuItem.connect('activate', function () {
-                Util.spawn(['gnome-control-center', 'info-overview']);
-            });
-
-            let devicesSettingsMenuItem = this._appendMenuItem(_('Device Management'));
-            devicesSettingsMenuItem.connect('activate', function () {
-                Util.spawn(['gnome-control-center', 'display']);
-            });
-
-            let disksMenuItem = this._appendMenuItem(_('Disk Management'));
-            disksMenuItem.connect('activate', function () {
-                Util.spawn(['gnome-disks']);
-            });
-
-            this._appendSeparator();
+            return null;
         }
+        
+        function _appendList(obj, cmd_list, title_list) {
+            if (cmd_list.length != title_list.length) {
+                return;
+            }
+            
+            for (var entry = 0; entry < cmd_list.length; entry++) {
```

Comment:
Old habits die hard? :P  Not to worry, but in js camelCase is normally preferred over snake_case.

Reply:
Nice catch! I switch languages way too often! By the way, I was wondering if there is a linter or tests I should be making sure to run? I was looking for contrib guidelines but didn't find any around this.

Reply:
I would be scared to run a linter on our code base :) We don't have any contribution guidelines atm, probably because we don't have much code contributions.